### ISSUE TITLE
gz-sim8 installation on macosx: libtiff.5.dylib not found

### ISSUE THREAD

**Issue by andreas-lehn** [conda-forge/gz-sim-feedstock#45]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

I installed `gz-sim8` and `gz-launch7` as described in https://community.gazebosim.org/t/gazebo-harmonic-packages-available-on-conda-forge/2330 into a freshly created conda environment. Starting `gz sim -s <my-world>` fails with a lot of error messages because some plug-ins can not be loaded.

The root cause for all these error messages is the same: `liftiff.5.dylib` cannot be loaded because it does not exist in the environment.

An example of such an error message is:

Error while loading the library [/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/gz-sim-8/plugins/libgz-sim-sensors-system.dylib]: dlopen(/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/gz-sim-8/plugins/libgz-sim-sensors-system.dylib, 0x0085): Library not loaded: @rpath/libtiff.5.dylib
  Referenced from: <E4C29E96-BF17-3672-B11D-6166FE92DAD0> /usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libproj.25.9.3.1.dylib
  Reason: tried: '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/gz-sim-8/plugins/../../libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/gz-sim-8/plugins/../../libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/lib/libtiff.5.dylib' (no such file), '/usr/local/Caskroom/miniforge/base/envs/gz-harmonic/bin/../lib/libtiff.5.dylib' (no such file), '/usr/local/lib/libtiff.5.dylib' (no such file), '/usr/lib/libtiff.5.dylib' (no such file, not in dyld cache)

The error message is correct: I checked all the typical location for libs on my Mac. There is no `libtiff.5.dylib` anywhere. I found `libtiff.dylib` which is a symbolic link to `liftiff.6.dylib`.  But no `liftiff.5.dylib`.

### Installed packages

```shell
aom                       3.8.1                h73e2aa4_0    conda-forge
assimp                    5.3.1                h460e769_2    conda-forge
azure-core-cpp            1.10.3               hbb1e571_1    conda-forge
azure-storage-blobs-cpp   12.10.0              he51d815_0    conda-forge
azure-storage-common-cpp  12.5.0               hf4badfb_2    conda-forge
blas                      1.0                    openblas  
blosc                     1.21.5               heccf04b_0    conda-forge
bullet-cpp                3.25                 hb19d194_2    conda-forge
bzip2                     1.0.8                h1de35cc_0  
c-ares                    1.26.0               h10d778d_0    conda-forge
ca-certificates           2023.12.12           hecd8cb5_0  
cairo                     1.18.0               h99e66fa_0    conda-forge
cfitsio                   4.3.1                h60fb419_0    conda-forge
cli11                     2.1.2                he9d5cce_0  
console_bridge            1.0.2                hbb4e6a2_1    conda-forge
cppzmq                    4.10.0               h0371a62_1    conda-forge
dartsim                   6.13.1               hba7a35e_2    conda-forge
dav1d                     1.2.1                h6c40b1e_0  
eigen                     3.4.0                ha357a0b_0  
expat                     2.5.0                hcec6c5f_0  
fcl                       0.7.0                h861072a_4    conda-forge
ffmpeg                    6.1.1           gpl_h73cf981_104    conda-forge
flann                     1.9.2                h3d5753d_0    conda-forge
fmt                       10.2.1               h7728843_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hd3eb1b0_0  
font-ttf-inconsolata      2.001                hcb22688_0  
font-ttf-source-code-pro  2.030                hd3eb1b0_0  
font-ttf-ubuntu           0.83                 h8b1ccd4_0  
fontconfig                2.14.2               h5bb23bf_0    conda-forge
fonts-anaconda            1                    h8fa9717_0  
fonts-conda-ecosystem     1                    hd3eb1b0_0  
freeimage                 3.18.0              h42008e6_18    conda-forge
freetype                  2.12.1               hd8bbffd_0  
freexl                    2.0.0                hce1ea5d_0  
fribidi                   1.0.10               haf1e3a3_0  
gdbm                      1.18                 hdccc71a_4  
geos                      3.12.1               h93d8f39_0    conda-forge
geotiff                   1.7.1               h509af15_15    conda-forge
gettext                   0.21.1               h8a4c099_0    conda-forge
giflib                    5.2.1                h6c40b1e_3  
glib                      2.78.3               hf4d7fad_0    conda-forge
glib-tools                2.78.3               hf4d7fad_0    conda-forge
gmp                       6.3.0                h93d8f39_0    conda-forge
gnutls                    3.7.9                h1951705_0    conda-forge
graphite2                 1.3.14               he9d5cce_1  
gst-plugins-base          1.22.9               h3fb38fc_0    conda-forge
gstreamer                 1.22.9               hf63bbb8_0    conda-forge
gts                       0.7.6                h6759243_3  
gz-launch7                7.0.0                h694c41f_4    conda-forge
gz-sim8                   8.0.0                h694c41f_5    conda-forge
gz-sim8-python            8.0.0            py38hba52a18_5    conda-forge
harfbuzz                  8.3.0                hf45c392_0    conda-forge
hdf4                      4.2.15               h8138101_7    conda-forge
hdf5                      1.14.3          nompi_h691f4bf_100    conda-forge
icu                       73.2                 hf5e326d_0    conda-forge
imath                     3.1.9                h7d26f99_0    conda-forge
json-c                    0.17                 h8e11ae5_0    conda-forge
jsoncpp                   1.9.5                h940c156_1    conda-forge
jxrlib                    1.1                  haf1e3a3_2  
kealib                    1.5.3                h5f07ac3_0    conda-forge
krb5                      1.21.2               hb884880_0    conda-forge
lame                      3.100                h1de35cc_0  
lcms2                     2.16                 ha2f27b4_0    conda-forge
lerc                      4.0.0                hb486fe8_0    conda-forge
libabseil                 20230802.1      cxx17_h048a20a_0    conda-forge
libaec                    1.1.2                he965462_1    conda-forge
libarchive                3.7.2                hd35d340_1    conda-forge
libass                    0.17.1               h80904bb_1    conda-forge
libboost                  1.82.0               hf53b9f2_2  
libccd-double             2.1                  he965462_3    conda-forge
libclang                  15.0.7          default_h6b1ee41_4    conda-forge
libclang13                15.0.7          default_h89cd682_4    conda-forge
libcrc32c                 1.1.2                he9d5cce_0  
libcurl                   8.5.0                h726d00d_0    conda-forge
libcxx                    16.0.6               hd57cbcb_0    conda-forge
libdeflate                1.19                 ha4e1b8e_0    conda-forge
libedit                   3.1.20230828         h6c40b1e_0  
libev                     4.33                 h9ed2024_1  
libexpat                  2.5.0                hf0c8a7f_1    conda-forge
libffi                    3.4.4                hecd8cb5_0  
libgdal                   3.8.3                hf248df1_2    conda-forge
libgfortran               5.0.0           13_2_0_h97931a8_2    conda-forge
libgfortran5              13.2.0               h2873a65_2    conda-forge
libglib                   2.78.3               h198397b_0    conda-forge
libgoogle-cloud           2.12.0               he77a663_5    conda-forge
libgrpc                   1.60.0               h038e8f1_1    conda-forge
libgz-cmake3              3.4.1                he965462_0    conda-forge
libgz-common5             5.5.0                h3a4a2cd_0    conda-forge
libgz-fuel-tools9         9.0.0                ha55963c_4    conda-forge
libgz-gui8                8.0.0                hbdce014_4    conda-forge
libgz-launch7             7.0.0                h5cf5aa8_4    conda-forge
libgz-math7               7.3.0                he965462_1    conda-forge
libgz-msgs10              10.1.0               ha919c77_0    conda-forge
libgz-msgs9               9.5.0                h9ccdae9_5    conda-forge
libgz-physics7            7.0.0                h5bfaae2_2    conda-forge
libgz-plugin2             2.0.2                he965462_0    conda-forge
libgz-rendering7          7.4.2                h93d8f39_0    conda-forge
libgz-rendering8          8.0.0                he965462_0    conda-forge
libgz-sensors7            7.3.0                h6fb071d_3    conda-forge
libgz-sensors8            8.0.0                h6fb071d_3    conda-forge
libgz-sim8                8.0.0                h85a05ee_5    conda-forge
libgz-tools2              2.0.0                h4425e3a_4    conda-forge
libgz-transport12         12.2.1               heb4d3be_5    conda-forge
libgz-transport13         13.0.0               hb6d2b27_5    conda-forge
libgz-utils2              2.1.0                he965462_0    conda-forge
libhwloc                  2.9.3           default_h24e0189_1009    conda-forge
libiconv                  1.17                 hd75f5a5_2    conda-forge
libidn2                   2.3.4                h6c40b1e_0  
libjpeg-turbo             3.0.0                h0dc2134_1    conda-forge
libkml                    1.3.0                h85bf17e_7  
libllvm15                 15.0.7               hbedff68_4    conda-forge
libnetcdf                 4.9.2           nompi_h7760872_113    conda-forge
libnghttp2                1.58.0               h64cf6d3_1    conda-forge
libode                    0.16.2              h940360d_12    conda-forge
libogg                    1.3.5                hca72f7f_1  
libopenblas               0.3.21               h54e7dc3_0  
libopenvino               2023.3.0             h113ac47_0    conda-forge
libopenvino-auto-batch-plugin 2023.3.0             h9adb129_0    conda-forge
libopenvino-auto-plugin   2023.3.0             h9adb129_0    conda-forge
libopenvino-hetero-plugin 2023.3.0             hfe2fe54_0    conda-forge
libopenvino-intel-cpu-plugin 2023.3.0             h113ac47_0    conda-forge
libopenvino-ir-frontend   2023.3.0             hfe2fe54_0    conda-forge
libopenvino-onnx-frontend 2023.3.0             hd0b7f58_0    conda-forge
libopenvino-paddle-frontend 2023.3.0             hd0b7f58_0    conda-forge
libopenvino-pytorch-frontend 2023.3.0             hd427752_0    conda-forge
libopenvino-tensorflow-frontend 2023.3.0             h35b5a9d_0    conda-forge
libopenvino-tensorflow-lite-frontend 2023.3.0             hd427752_0    conda-forge
libopus                   1.3.1                h1de35cc_0  
libpng                    1.6.42               h92b6c6a_0    conda-forge
libpq                     16.1                 ha925e61_7    conda-forge
libprotobuf               4.25.1               hc4f2305_1    conda-forge
libraw                    0.21.1               h8138101_2    conda-forge
libre2-11                 2023.06.02           h4694dbf_0    conda-forge
librttopo                 1.1.0               hf05f67e_15    conda-forge
libsdformat13             13.6.0               h99302ca_4    conda-forge
libsdformat14             14.0.0               h8ab32a3_2    conda-forge
libsodium                 1.0.18               h1de35cc_0  
libspatialite             5.1.0                hebe6af1_4    conda-forge
libsqlite                 3.44.2               h92b6c6a_0    conda-forge
libssh2                   1.11.0               hd019ec5_0    conda-forge
libtasn1                  4.19.0               h6c40b1e_0  
libtiff                   4.6.0                h684deea_2    conda-forge
libunistring              0.9.10               h9ed2024_0  
libvorbis                 1.3.7                haf1e3a3_0  
libvpx                    1.13.1               hcec6c5f_0  
libwebp-base              1.3.2                h6c40b1e_0  
libxcb                    1.15                 hb7f2c08_0    conda-forge
libxml2                   2.12.5               hc0ae0f7_0    conda-forge
libzip                    1.10.1               hc158999_3    conda-forge
libzlib                   1.2.13               h8a1eda9_5    conda-forge
llvm-openmp               17.0.6               hb6ac08f_0    conda-forge
lz4-c                     1.9.4                hcec6c5f_0  
lzo                       2.10                 haf1e3a3_2  
minizip                   4.0.3                h79ad51c_0  
mysql-common              8.0.33               h1d20c9b_6    conda-forge
mysql-libs                8.0.33               hed35180_6    conda-forge
ncurses                   6.4                  hcec6c5f_0  
nettle                    3.9.1                h8e11ae5_0    conda-forge
nspr                      4.35                 hcec6c5f_0  
nss                       3.97                 ha05da47_0    conda-forge
numpy                     1.24.3           py38h57a7bef_0  
numpy-base                1.24.3           py38hc93c6d9_0  
octomap                   1.9.8                hb8565cd_0    conda-forge
ogre                      1.10.12             h32f507b_16    conda-forge
ogre-next                 2.3.1                h7603aad_4    conda-forge
openexr                   3.2.1                h747cbf1_0    conda-forge
openh264                  2.4.1                h73e2aa4_0    conda-forge
openjpeg                  2.5.0                ha4da562_3    conda-forge
openssl                   3.2.1                hd75f5a5_0    conda-forge
p11-kit                   0.24.1               h65f8906_0    conda-forge
pcre                      8.45                 h23ab428_0  
pcre2                     10.42                h9b97e30_0  
pip                       23.3.1           py38hecd8cb5_0  
pixman                    0.43.2               h73e2aa4_0    conda-forge
poppler                   24.02.0              h0c752f9_0    conda-forge
poppler-data              0.4.11               hecd8cb5_1  
postgresql                16.1                 hbd19fd8_7    conda-forge
proj                      9.3.1                h1972728_0  
pthread-stubs             0.3                  hdd91f34_1  
pugixml                   1.14                 he965462_0    conda-forge
pybind11-abi              4                    hd3eb1b0_1  
python                    3.8.18          h5ba8234_1_cpython    conda-forge
python_abi                3.8                      4_cp38    conda-forge
qt-main                   5.15.8              h26cf8db_18    conda-forge
re2                       2023.06.02           hd34609a_0    conda-forge
readline                  8.2                  hca72f7f_0  
ruby                      3.2.2                he8a3799_0  
sdl2                      2.28.5               h73e2aa4_1    conda-forge
setuptools                68.2.2           py38hecd8cb5_0  
snappy                    1.1.10               hcec6c5f_1  
spdlog                    1.12.0               h8dd852c_2    conda-forge
sqlite                    3.41.2               h6c40b1e_0  
svt-av1                   1.8.0                h93d8f39_0    conda-forge
swig                      4.0.2                he9d5cce_4  
tbb                       2021.11.0            h7728843_1    conda-forge
tiledb                    2.19.1               h8fd0293_0    conda-forge
tinyxml2                  10.0.0               h73e2aa4_0    conda-forge
tk                        8.6.13               h1abcd95_1    conda-forge
tzcode                    2024a                h10d778d_0    conda-forge
tzdata                    2023d                h04d1e81_0  
urdfdom                   4.0.0                h760acac_1    conda-forge
urdfdom_headers           1.1.1                h7728843_0    conda-forge
uriparser                 0.9.7                h6c40b1e_0  
wheel                     0.41.2           py38hecd8cb5_0  
x264                      1!164.3095           h775f41a_2    conda-forge
x265                      3.5                  hbb4e6a2_3    conda-forge
xerces-c                  3.2.5                hbbe9ea5_0    conda-forge
xorg-kbproto              1.0.7             h35c211d_1002    conda-forge
xorg-libice               1.1.1                h0dc2134_0    conda-forge
xorg-libsm                1.2.4                h0dc2134_0    conda-forge
xorg-libx11               1.8.7                hbd0b022_0    conda-forge
xorg-libxau               1.0.11               h0dc2134_0    conda-forge
xorg-libxaw               1.0.14               h0d85af4_1    conda-forge
xorg-libxdmcp             1.1.3                h35c211d_0    conda-forge
xorg-libxext              1.3.4                hb7f2c08_2    conda-forge
xorg-libxmu               1.1.3                h0d85af4_0    conda-forge
xorg-libxpm               3.5.17               h0dc2134_0    conda-forge
xorg-libxt                1.3.0                h0dc2134_1    conda-forge
xorg-xextproto            7.3.0             hb7f2c08_1003    conda-forge
xorg-xproto               7.0.31            h9ed2024_1007  
xz                        5.4.5                h6c40b1e_0  
yaml                      0.2.5                haf1e3a3_0  
zeromq                    4.3.5                hcec6c5f_0  
zlib                      1.2.13               h8a1eda9_5    conda-forge
zstd                      1.5.5                hc035e20_0  
zziplib                   0.13.69              h5dbffcc_1    conda-forge
```

### Environment info

```shell
active environment : gz-harmonic
    active env location : /usr/local/Caskroom/miniforge/base/envs/gz-harmonic
            shell level : 4
       user config file : /Users/lae2fr/.condarc
 populated config files : /usr/local/Caskroom/miniforge/base/.condarc
                          /Users/lae2fr/.condarc
          conda version : 23.11.0
    conda-build version : not installed
         python version : 3.10.13.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=skylake
                          __conda=23.11.0=0
                          __osx=13.6.1=0
                          __unix=0=0
       base environment : /usr/local/Caskroom/miniforge/base  (writable)
      conda av data dir : /usr/local/Caskroom/miniforge/base/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /usr/local/Caskroom/miniforge/base/pkgs
                          /Users/lae2fr/.conda/pkgs
       envs directories : /usr/local/Caskroom/miniforge/base/envs
                          /Users/lae2fr/.conda/envs
               platform : osx-64
             user-agent : conda/23.11.0 requests/2.31.0 CPython/3.10.13 Darwin/22.6.0 OSX/13.6.1 solver/libmamba conda-libmamba-solver/23.12.0 libmambapy/1.5.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

**Comment by traversaro** [comment]
@andreas-lehn It seems that for some reason in your environment you have a mix of packages from default/pkgs and conda-forge, and that is probably the problem. Can you try to create a fresh environment, i.e. :

~~~
conda create -n gazeboharmonic -c conda-forge gz-sim8 gz-launch7
~~~

and check again? Probably it could make sense to document explicitly this in https://community.gazebosim.org/t/gazebo-harmonic-packages-available-on-conda-forge/2330 .

**Comment by andreas-lehn** [comment]
@traversaro I did exactly what you requested and installed `gz-sim8` and `gz-launch7` again in a newly created environment and it worked without any problems! This is weird because I thought I did exactly the same before. Nevertheless my environment now contains only packages from conda-forge and no packages from defaults. I think your suggestion is right that the mix of packages was the reason why it did not work. I have no glue how this happened.
